lib/xlibc-rt/tls.c

```diff
@@ -0,0 +1,20 @@
+#include "xboxkrnl/xboxkrnl.h"
+#include "_tls.h"
+
+ULONG _tls_index = 0;
+
+#pragma data_seg(".tls")
+char _tls_start = 0;
+#pragma data_seg("tls$ZZZ")
+char _tls_end = 0;
```

Comment:
Does our linker script (which one do we even use?) guarantee preservation of this section ordering? What's the "$ZZZ" suffix? Why isn't it ".tls$ZZZ"? Do you have any docs for this?

Also, does cxbe preserve these sections (or do they get merged elsewhere)?

Reply:
> Why isn't it ".tls$ZZZ"?

It should be ".tls$ZZZ", the missing dot is a typo. Interesting that it still worked, though.

> What's the "$ZZZ" suffix?

The "$ZZZ" suffix is a way to put stuff at the end of a section.
I can't find MS documentation about it right now, but [this](http://www.nynaeve.net/?p=183) contains a description of the behavior near the end.

> Also, does cxbe preserve these sections (or do they get merged elsewhere)?

When inspecting the exe-file, it seems that lld merges the sections (into rdata, I assume), which is fine, since we really only need it to generate IMAGE_TLS_DIRECTORY.